src/Compilers/CSharp/Portable/CodeGen/EmitConversion.cs

```diff
@@ -143,8 +143,8 @@ private void EmitConversion(BoundConversion conversion)
 #if DEBUG
                     switch (fromPredefTypeKind)
                     {
-                        case Microsoft.Cci.PrimitiveTypeCode.IntPtr:
-                        case Microsoft.Cci.PrimitiveTypeCode.UIntPtr:
+                        case Microsoft.Cci.PrimitiveTypeCode.IntPtr when !fromType.IsNativeIntegerType:
```

Comment:
>when !fromType.IsNativeIntegerType [](start = 68, length = 34)

Should asserts in the ```default``` case be adjusted as well for extra validation around native integer types?